overlord/devicestate/devicemgr.go

```diff
@@ -395,16 +402,21 @@ func (m *DeviceManager) ensureUbuntuSaveSnapFolders() error {
 // setupUbuntuSave sets up ubuntu-save partition. It makes sure
 // to mount ubuntu-save (if feasible), and ensures the correct snap
 // folders are present according to currently installed snaps.
-func (m *DeviceManager) setupUbuntuSave() error {
+func (m *DeviceManager) setupUbuntuSave(dev snap.Device) error {
 	if err := m.ensureUbuntuSaveIsMounted(); err != nil {
 		return err
 	}
 
 	// At this point ubuntu-save should be available under the
 	// /var/lib/snapd/save path, so we mark the partition as such.
+	m.saveAvailable = true
+
 	// The last step is to ensure needed folder structure is present
 	// for the per-snap folder storage.
-	m.saveAvailable = true
+	// We support this only on Core for now.
+	if dev == nil || dev.Classic() {
```

Comment:
If you agree with the previous suggestion, this too should be just
```suggestion
	if dev.Classic() {
```